optuna/samplers/_tpe/sampler.py

```diff
@@ -294,14 +294,17 @@ def infer_relative_search_space(
             self._search_space_group = self._group_decomposed_search_space.calculate(study)
             for sub_space in self._search_space_group.search_spaces:
                 for name, distribution in sub_space.items():
-                    if not isinstance(distribution, _DISTRIBUTION_CLASSES):
+                    if (
+                        not isinstance(distribution, _DISTRIBUTION_CLASSES)
+                        or distribution.single()
```

Comment:
Do you think we should raise warning in this single case too? This is not a strong opinion but I think it's irrelevant to most users and perhaps leads to unnecessary caution. It might be useful for debugging, though.

Reply:
Totally agree with you. I changed not to raise warnings in the single-point distribution cases.